app/containers/Products/ProductCreatorContainer.tsx

```diff
@@ -1,51 +1,79 @@
 import React from 'react';
-import {createFragmentContainer} from 'react-relay';
-import {Form, Button, Col} from 'react-bootstrap';
+import {createFragmentContainer, RelayProp} from 'react-relay';
+import JsonSchemaForm, {IChangeEvent} from 'react-jsonschema-form';
+import {JSONSchema6} from 'json-schema';
+import {Button, Card, Collapse} from 'react-bootstrap';
+import FormArrayFieldTemplate from '../Forms/FormArrayFieldTemplate';
+import FormFieldTemplate from '../Forms/FormFieldTemplate';
+import FormObjectFieldTemplate from '../Forms/FormObjectFieldTemplate';
 import createProductMutation from '../../mutations/product/createProductMutation';
 
-export const ProductCreator = props => {
-  const createProductFromRef = React.createRef<Form & HTMLFormElement>();
+interface Props {
+  relay: RelayProp;
+  expanded: boolean;
+  resetForm: (...args: any[]) => void;
+  createProductFormKey: number;
+}
 
-  const saveProduct = async event => {
-    event.preventDefault();
-    event.stopPropagation();
-    const date = new Date().toUTCString();
+export const ProductCreator: React.FunctionComponent<Props> = ({
+  relay,
+  expanded,
+  resetForm,
+  createProductFormKey
+}) => {
+  const saveProduct = async (e: IChangeEvent) => {
     const variables = {
       input: {
         product: {
-          name: event.target.product_name.value,
-          description: event.target.product_description.value,
+          name: e.formData.product,
+          description: e.formData.description,
           state: 'active',
-          parent: [null],
-          createdAt: date,
-          createdBy: 'Admin'
+          parent: [null]
         }
       }
     };
-
-    const {environment} = props.relay;
+    resetForm();
+    const {environment} = relay;
     await createProductMutation(environment, variables);
-    createProductFromRef.current.reset();
+  };
+
+  // Schema for JsonSchemaForm component
+  const createProductSchema: JSONSchema6 = {
```

Comment:
You can move this variable outside of the component, so that the schema prop doesn't change for every render